lnwallet/wallet.go

```diff
@@ -1214,6 +1215,28 @@ func (l *LightningWallet) initOurContribution(reservation *ChannelReservation,
 	reservation.partialState.RevocationProducer = producer
 	reservation.ourContribution.ChannelConstraints = l.Cfg.DefaultConstraints
 
+	// If taproot channels are active, then we'll generate two sets of
+	// nonces: one for our local commitment, and one for their remote
+	// commitment.
+	if reservation.partialState.ChanType.IsTaproot() {
+		// As we'd like the local nonce we send over to be generated
+		// determinstically, we'll provide a custom reader that
+		// actually just uses our sha-chain pre-image as the primary
+		// randomness source.
```

Comment:
So the danger is signing with the same nonce twice with a different messag, this is our local nonce, which we'll use only to sign for our _local_ commitment state. We only sign for this commitment state once we go to broadcast, which only happens on force close. 

The nonce we want to be careful about is the nonce we send to the remote party to sign their remote commitment transaction. If we reuse this nonce (lets say we didn't generate a new nonce JIT style for transmissions), then we'd be in danger.

Reply:
Chatted with @Crypt-iQ here, and decided we need to have this be a _distinct_ root, as otherwise they'll learn this from the very first revoke message. 

The new approach creates a new shachain root via hmac from the old root, then uses that in place. A party never learns our shachain root unless all 2^48-1 states are exhausted.